pkg/controllers/user/certsexpiration/certsexpiration.go

```diff
@@ -161,3 +164,39 @@ func getCertExpiration(c string) (v3.CertExpiration, error) {
 		ExpirationDate: date.Format(time.RFC3339),
 	}, nil
 }
+
+//deleteUnusedCerts removes unused certs and cleans up kubelet certs when GenerateServingCertificate is disabled
+func deleteUnusedCerts(certsExpInfo map[string]v3.CertExpiration, cluster *v3.Cluster) {
+	logrus.Info("Deleting unused certificates")
+	unusedCerts := make(map[string]bool)
+	for k := range certsExpInfo {
+		if strings.HasPrefix(k, pki.EtcdCertName) || strings.HasPrefix(k, pki.KubeletCertName) {
+			unusedCerts[k] = true
+		}
+	}
+	etcdHosts := hosts.NodesToHosts(cluster.Status.AppliedSpec.RancherKubernetesEngineConfig.Nodes, services.ETCDRole)
```

Comment:
Do we need a nil check here just in case? I only found the `cluster.Spec.RancherKubernetesEngineConfig` in `sync`. If this is caught else where, that's fine.

Reply:
Updated.